===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(rpc_exec_init:argv->elem)) at 24 (line 1053, column 2),
        Insert: if (true) free(*(rpc_exec_init:argv)) at 25 (line 1053, column 2) }
===== Actual patch in diff format =====
@@ -1050,7 +1050,12 @@ rpc_exec_init (const char *remote,
 	}
 
 	rex = calloc (1, sizeof (rpc_exec));
-	return_val_if_fail (rex != NULL, NULL);
+	if (rex != NULL) { } else {
+		p11_debug_precond ("p11-kit: '%s' not true at %s\n", #x, __func__);
+		free(argv->elem);
+		free(argv);
+		return NULL;
+	}
 
 	p11_array_push (argv, NULL);
 	rex->argv = argv;
